src/pages/BuyDomain/BuyDomain.tsx

```diff
@@ -0,0 +1,70 @@
+import type { FC } from 'react';
+
+import { useState } from 'react';
+import {
+	IconZero,
+	DomainSearch,
+	BuyDomainModal,
+	ConnectWallet,
+} from '../../components';
+import { useWeb3 } from '../../hooks';
+import styles from './BuyDomain.module.scss';
+
+export const BuyDomain: FC = () => {
+	const { account } = useWeb3();
+
+	const [buyDomainModal, setBuyDomainModal] = useState<{
+		isOpen;
+		domainName;
+	}>({
+		isOpen: false,
+		domainName: '',
+	});
+
+	const handleOnBuyButtonClick = (domainName) => {
+		setBuyDomainModal({
+			isOpen: true,
+			domainName,
+		});
+	};
+
+	const handleOnModalClose = () => {
+		setBuyDomainModal({
+			...buyDomainModal,
+			isOpen: false,
+		});
+	};
+
+	if (!account) {
+		return (
+			<div className={styles.Container}>
+				<div className={styles.Wrapper}>
+					<ConnectWallet
+						message={'Connect a Web3 wallet to see your Buy Domains data.'}
+					/>
+				</div>
+			</div>
+		);
+	}
```

Comment:
I have had this pattern appear in my code too - maybe we should make a wrapper in zUI which prevents children rendering if the user hasn't connected their wallet. For now this is all good!